api/repositories/build_repository_test.go

```diff
@@ -511,6 +511,114 @@ var _ = Describe("BuildRepository", func() {
 			})
 		})
 	})
+
+	Describe("ListBuilds", func() {
+		const (
+			app1GUID      = "app-1-guid"
+			app2GUID      = "app-2-guid"
+			package1GUID  = "package-1-guid"
+			package2GUID  = "package-2-guid"
+			stagingMemory = 1024
+			stagingDisk   = 2048
+		)
+
+		var (
+			namespace1 *korifiv1alpha1.CFSpace
+			namespace2 *korifiv1alpha1.CFSpace
+			cfOrg      *korifiv1alpha1.CFOrg
+		)
+
+		BeforeEach(func() {
+			cfOrg = createOrgWithCleanup(ctx, prefixedGUID("org"))
+			namespace1 = createSpaceWithCleanup(ctx, cfOrg.Name, prefixedGUID("space2"))
+			namespace2 = createSpaceWithCleanup(ctx, cfOrg.Name, prefixedGUID("space3"))
+		})
+
+		AfterEach(func() {
+			Expect(k8sClient.Delete(ctx, namespace1)).To(Succeed())
+			Expect(k8sClient.Delete(ctx, namespace2)).To(Succeed())
+		})
+
+		makeBuild := func(namespace, buildGUID, packageGUID, appGUID string) *korifiv1alpha1.CFBuild {
+			return &korifiv1alpha1.CFBuild{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      buildGUID,
+					Namespace: namespace,
+					Labels: map[string]string{
+						korifiv1alpha1.SpaceGUIDKey: namespace,
+					},
+				},
+				Spec: korifiv1alpha1.CFBuildSpec{
+					PackageRef: corev1.LocalObjectReference{
+						Name: packageGUID,
+					},
+					AppRef: corev1.LocalObjectReference{
+						Name: appGUID,
+					},
+					StagingMemoryMB: stagingMemory,
+					StagingDiskMB:   stagingDisk,
+					Lifecycle: korifiv1alpha1.Lifecycle{
+						Type: "buildpack",
+						Data: korifiv1alpha1.LifecycleData{
+							Buildpacks: []string{},
+							Stack:      "",
+						},
+					},
+				},
+			}
+		}
+
+		When("on the happy path", func() {
+			var (
+				build1GUID string
+				build2GUID string
+				build1     *korifiv1alpha1.CFBuild
+				build2     *korifiv1alpha1.CFBuild
+			)
+
+			BeforeEach(func() {
+				build1GUID = uuid.NewString()
+				build2GUID = uuid.NewString()
+				build1 = makeBuild(namespace1.Name, build1GUID, package1GUID, app1GUID)
+				Expect(k8sClient.Create(ctx, build1)).To(Succeed())
+				build2 = makeBuild(namespace2.Name, build2GUID, package2GUID, app2GUID)
+				Expect(k8sClient.Create(ctx, build2)).To(Succeed())
+
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, namespace1.Name)
+				createRoleBinding(ctx, userName, spaceDeveloperRole.Name, namespace2.Name)
+			})
+
+			When("listing all builds", func() {
```

Comment:
This context does not alter the default setup, therefore it makes no sense (similarly to the "on the happy path" context)